appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java

```diff
@@ -220,35 +196,50 @@ public void registerClassLoader(String appName, ClassLoader classloader) {
             }
         }
     }
+    
+    private Map<String, Set<HealthCheck>> getHealthChecks(HealthCheckType type) {
+        final Map<String, Set<HealthCheck>> healthChecks;
+        if (type == READINESS) {
+            healthChecks = readiness;
+        } else if (type == LIVENESS) {
+            healthChecks = liveness;
+        } else {
+            healthChecks = health;
+        }
+        return healthChecks;
+    }
+    
+        private Map<String, Set<HealthCheck>> getCollectiveHealthChecks(HealthCheckType type){
+        final Map<String, Set<HealthCheck>> healthChecks;
+        if (type == READINESS) {
+            healthChecks = readiness;
+        } else if (type == LIVENESS) {
+            healthChecks = liveness;
+        } else {
+            healthChecks = new HashMap<>(health);
```

Comment:
What I understand from this now is that when running health checks when doing type `READINESS` or `LIVENESS` we run just those but for `HEALTH` we run any kind of check? Is that it?

Reply:
Yes @jbee, `HEALTH` is deprecated hence merging all other health checks in output.

> The MicroProfile Health Check architecture consists of two /health/ready and /health/live endpoints in a MicroProfile runtime that respectively represent the readiness and the liveness of the entire runtime.
For backward compatibility, a 3rd endpoint /health may also be used to provide a combination of previous endpoints and Health Check procedures annotated with the deprecated @Health annotation.

https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/overview.adoc

Reply:
Thanks @jGauravGupta for the explanation. Couldn't we keep all checks in a common data structure and process them by using a filter? I'd think that this gets much easier to manage a single collection and to send a filter predicate to the processing - something like `(check) -> check.getType() == READINESS` and for backwards compatibility it is just `(check) -> true`.

Reply:
One health check could also be part of more than one type e.g both `@Readiness` and `@Liveness`. 
Also, Health Check endpoints are invoked more frequently and intended for the machine to machine communication, Removing type-based filter (and iteration) optimized the request further.

Reply:
I don't see how constructing a new collection each time is more optimised then iterating the same one and applying a filter. Especially when a check can be of more then one type the filter seems a good approach to me. Still, it is just a suggestion.

Reply:
> how constructing a new collection each time is more optimised
But this case is specifically for deprecated `/health` endpoint, which is not recommended to use.